### DISCUSSION TITLE
Use non-english letters in values of fields inside queries/mutations

### DISCUSSION THREAD

**Discussion by arkadyz1** [graphql-java/graphql-java#2794] [Q&A] [answered]
Hi, team

We are trying to make mutations (from Apollo client and from graphiQL client) which contain Hebrew values inside the fields, for example:
"address": "שלום.com"

And we see in arguments of DataFetcher the value of "address" is coming broken. We are using graphQL version 14.1

Can anyone help ?

Thank you,
Arkady

**Comment by jord1e** [comment]
Hey Akady,

```java
TypeDefinitionRegistry tdr = new SchemaParser().parse("type Query { a: String! }");

RuntimeWiring rtw =
    RuntimeWiring.newRuntimeWiring()
        .type("Query", queryEnv -> queryEnv.dataFetcher("a", env -> "שלום.com"))
        .build();

GraphQLSchema schema = new SchemaGenerator().makeExecutableSchema(tdr, rtw);

GraphQL graphQL = GraphQL.newGraphQL(schema).build();

System.out.println(
    ((Map<?, ?>) graphQL.execute("{ a }").toSpecification().get("data"))
        .get("a")
        .equals("שלום.com"));
```

outputs `true` for me. What web framework are you using? DGS Framework, Spring for GraphQL, graphql-java-kickstart, graphql-kotlin?

**Reply by arkadyz1** [reply]
We are using graphql-java-kickstart-9.2.0.jar, graphql-java-servlet-9.2.0.jar

**Reply by jord1e** [reply]
Then you should open a issue there, there is probably a encoding issue (have you setting the correct headers? https://www.w3.org/International/articles/http-charset/index):

https://github.com/graphql-java-kickstart/graphql-spring-boot
https://github.com/graphql-java-kickstart/graphql-java-servlet

**Reply by bbakerman** [reply]
This is nominally a kickstart issue - but also note in later versions of graphql-java (17.0) we did more work on correct unicode encoding

https://github.com/graphql-java/graphql-java/pull/2404

**Comment by arkadyz1** [comment, marked as answer]
Upgrading kikstart and servlet to 12.0.0  + graphQL java to 17.3 didn't help. 

I found the problem. We are using Wildfly 23 as application server and its HttpServletRequestImpl uses "ISO_8859_1" as default encoding.

When I wrote following function in my implementation of GraphQLHttpServlet so the problem was solved.

	public void doPost(HttpServletRequest req, HttpServletResponse resp) {
		try {
			req.setCharacterEncoding("UTF-8");
		} catch (UnsupportedEncodingException e) {
		}
		super.doPost(req, resp);
	}
